capsules/extra/src/nmea.rs

```diff
@@ -0,0 +1,155 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
+
+//! Provides userspace with access to NMEA sentences.
+
+use kernel::errorcode::into_statuscode;
+use kernel::grant::{AllowRoCount, AllowRwCount, Grant, UpcallCount};
+use kernel::processbuffer::WriteableProcessBuffer;
+use kernel::syscall::{CommandReturn, SyscallDriver};
+use kernel::utilities::cells::TakeCell;
+use kernel::{ErrorCode, ProcessId};
+
+/// Client for receiving NMEA sentences
+pub trait NmeaClient {
+    /// Called when a full NMEA sentence has been found, or if an error
+    /// occurs.
+    ///
+    /// The possible errors are:
+    /// - `BUSY`: Indicates that the hardware is busy with an existing
+    ///           operation or initialisation/calibration.
+    /// - `NOACK`: Failed to correctly communicate over communication protocol.
+    fn callback(&self, buffer: &'static mut [u8], len: usize, status: Result<(), ErrorCode>);
```

Comment:
You're also returning `NOSUPPORT` in the case where the NMEA sentence was not valid UTF8.